src/Mvc/Mvc.Abstractions/src/ModelBinding/BindingSource.cs

```diff
@@ -86,6 +86,15 @@ public class BindingSource : IEquatable<BindingSource?>
         isGreedy: true,
         isFromRequest: false);
 
+    /// <summary>
+    /// A <see cref="BindingSource"/> for request keyed services.
+    /// </summary>
+    public static readonly BindingSource KeyedServices = new BindingSource(
```

Comment:
I was hoping that we could reuse the `Services` binding source to avoid having to introduce new API here. It would require that we pull out the `FromKeyedServices` attribute again in the model binder but we'd be introducing less new API space.

Thoughts?

Reply:
The issue is for keyed services, we need to do some reflection to extract the key. I definitely don't want that when it's not necessary.

If there is an elegant way to pass the key from `BindingInfo` to the `IModelBinder` or `IModelBinderProvider` that would be perfect.

Reply:
If it is better to add a property to `BindingInfo`, then it's definitly easier.